code/game/objects/structures/ladders.dm

```diff
@@ -1,48 +1,55 @@
+// Basic ladder. By default links to the z-level above/below.
 /obj/structure/ladder
 	name = "ladder"
 	desc = "A sturdy metal ladder."
 	icon = 'icons/obj/structures.dmi'
 	icon_state = "ladder11"
-	var/id = null
-	var/height = 0							//the 'height' of the ladder. higher numbers are considered physically higher
+	anchored = TRUE
 	var/obj/structure/ladder/down = null	//the ladder below this one
 	var/obj/structure/ladder/up = null		//the ladder above this one
-	var/auto_connect = FALSE
-
-/obj/structure/ladder/unbreakable //mostly useful for awaymissions to prevent halting progress in a mission
-	name = "sturdy ladder"
-	desc = "An extremely sturdy metal ladder."
 
-
-/obj/structure/ladder/Initialize(mapload)
-	GLOB.ladders += src
+/obj/structure/ladder/Initialize(mapload, obj/structure/ladder/up=null, obj/structure/ladder/down=null)

```

Comment:
remove the `= null` from args(it already defaults to null when a var is initialized)